macros/parsers/parserLinearRelation.pl

```diff
@@ -0,0 +1,386 @@
+#    $R = LinearRelation([1,2,1], 5, "<=");
+
+sub new {
+	my ($self, $N, $p, $bop) = @_;
+	my $class   = ref($self) || $self;
+	my $context = (Value::isContext($_[0]) ? shift : $self->context);
+	my $formula = "Value::Formula";
+	return shift if scalar(@_) == 1 && ref($_[0]) eq 'LinearRelation';
+	#$_[0] = $context->Package("Vector")->new($context, $_[0]) if ref($_[0]) eq 'ARRAY';
+	#$_[1] = $context->Package("Point")->new($context, $_[1])  if ref($_[1]) eq 'ARRAY';
+
+	# 2 or more:
+	# first argument is normal vector, possibly a Vector, Point, or ARRAY ref
+	# second argument is point or number, possibly a Vector, Point, ARRAY ref, Real, or perl real
+	# optional third argument is symbol, defaults to '='
+
+	my ($plane, $d);
+
+	if (defined $p) {
+		# Make sure $N is a Vector
+		$N = $context->Package("Vector")->new($context, $N);
+		# Make sure $p is a Point or Real
+		$p = $context->Package("Point")->new($context, $p) if (ref($p) eq 'ARRAY' || ref($p) eq 'Value::Vector');
+		$p = $context->Package("Real")->new($context, $p)  if ref($p) ne 'Value::Point';
+		# Constant on the right side
+		$d = (ref($p) eq 'Value::Real') ? $p : $p . $N;
+		my @terms;
+		my $i = 0;
+		for my $x ($context->variables->names) { push @terms, $N->{data}[ $i++ ]->string . $x }
+		$bop      = '=' unless defined($bop);
+		$leftside = $formula->new(join(' + ', @terms))->reduce(@_);
+		$plane    = $formula->new($leftside . $bop . $d->string);
+	} else {
+		# Determine the normal vector and d value from the equation
+		$plane = $N;
+		$plane = $formula->new($context, $plane) unless Value::isValue($plane);
+		$vars  = [ $context->variables->names ];
+		Value::Error("Your formula doesn't look like a linear relation")
+			unless $plane->type eq 'Relation';
+		# Find the coefficients of the formula
+		my $f = ($formula->new($context, $plane->{tree}{lop}) - $formula->new($context, $plane->{tree}{rop}))->reduce;
+		my $F = $f->perlFunction(undef, [ @{$vars} ]);
+		my @v = split('', '0' x scalar(@{$vars}));
+		$d = -&$F(@v);
+		my @coeff = (@v);
+		for my $i (0 .. scalar(@v) - 1) { $v[$i] = 1; $coeff[$i] = &$F(@v) + $d; $v[$i] = 0 }
+		$N = Value::Vector->new([@coeff]);
+		$plane =
+			$self->new($N, $d, $plane->{tree}{bop}, '-x=-y' => 0, '-x=n' => 0)
+			->with(original_formula => $plane, original_formula_latex => $plane->TeX);
+		# Check that the student's formula really is what we thought
+		Value::Error("Your formula isn't a linear one")
+			unless ($formula->new($plane->{tree}{lop}) - $formula->new($plane->{tree}{rop})) == $f;
+	}
+	$plane->{d} = $d;
+	$plane->{N} = $N;
+	return bless $plane, $class;
+}
+
+#
+#  If the vectors are zero, check if true or false
+#  If the vectors are non-zero, check if the equations are multiples of each other.
+#
+sub compare {
+	my ($self, $l, $r) = Value::checkOpOrder(@_);
+	$l = new LinearRelation($l) unless ref($l) eq ref($self);
+	$r = new LinearRelation($r) unless ref($r) eq ref($self);
```

Comment:
This should be changed to
```suggestion
	$l = LinearRelation->new($l) unless ref($l) eq ref($self);
	$r = LinearRelation->new($r) unless ref($r) eq ref($self);
```
See https://metacpan.org/pod/Perl::Critic::Policy::Objects::ProhibitIndirectSyntax